### PR TITLE
Always pass `freq` explicitly to `pd.period_range`.

### LINKED ISSUE #2063: pandas period_range does not infer the freq properly
## Description

In the dev branch:

```
testing[0][FieldName.START]
Period('2012-10-13 00:00', '30T')


pd.period_range(
                start=testing[0][FieldName.START],
                periods=testing[0][FieldName.TARGET].shape[-1],)

PeriodIndex(['2012-10-13 00:00', '2012-10-13 00:01', '2012-10-13 00:02',
             '2012-10-13 00:03', '2012-10-13 00:04', '2012-10-13 00:05',
             '2012-10-13 00:06', '2012-10-13 00:07', '2012-10-13 00:08',
             '2012-10-13 00:09',
             ...
             '2012-10-29 14:14', '2012-10-29 14:15', '2012-10-29 14:16',
             '2012-10-29 14:17', '2012-10-29 14:18', '2012-10-29 14:19',
             '2012-10-29 14:20', '2012-10-29 14:21', '2012-10-29 14:22',
             '2012-10-29 14:23'],
            dtype='period[30T]', length=23904)
```

increments every min.  👎🏽  However if I specify the freq its correct:

```
pd.period_range(
                start=testing[0][FieldName.START],
                periods=testing[0][FieldName.TARGET].shape[-1],
                freq=testing[0][FieldName.START].freq,
            )

PeriodIndex(['2012-10-13 00:00', '2012-10-13 00:30', '2012-10-13 01:00',
             '2012-10-13 01:30', '2012-10-13 02:00', '2012-10-13 02:30',
             '2012-10-13 03:00', '2012-10-13 03:30', '2012-10-13 04:00',
             '2012-10-13 04:30',
             ...
             '2014-02-22 19:00', '2014-02-22 19:30', '2014-02-22 20:00',
             '2014-02-22 20:30', '2014-02-22 21:00', '2014-02-22 21:30',
             '2014-02-22 22:00', '2014-02-22 22:30', '2014-02-22 23:00',
             '2014-02-22 23:30'],
            dtype='period[30T]', length=23904)
```

### PR DESCRIPTION
**Pull request by kashif**
fixes #2063 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by kashif** [comment]
opps I added the sec freq support here too...

File: src/gluonts/time_feature/__init__.py [hunk 1]
```diff
@@ -25,11 +25,13 @@
     MinuteOfHourIndex,
     MonthOfYear,
     MonthOfYearIndex,
-    norm_freq_str,
-    time_features_from_frequency_str,
+    SecondOfMinute,
+    SecondOfMinuteIndex,
```
Comment (jaheba) [inline src/gluonts/time_feature/__init__.py]: This looks unrelated.

Comment (kashif) [reply src/gluonts/time_feature/__init__.py]: ok let me close this and send just the fixes... sorry about that!

**Comment by jaheba** [comment]
Thanks, did you check all invocations to `pd.period_range`?

Alternatively, we could a helper which does this for us, then it's also easier to check that it's used.

**Comment by kashif** [comment]
yes i checked all the period_range

File: src/gluonts/time_feature/__init__.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/time_feature/__init__.py]: Can't you just update this PR with the relevant fixes?

Comment (kashif) [reply src/gluonts/time_feature/__init__.py]: and revert the sec. support commit?

Comment (jaheba) [reply src/gluonts/time_feature/__init__.py]: What I mean is something like this:

```sh
git checkout -b sec-freq
git checkout -
<revert changes>
git push
```

**Comment by jaheba** [comment]
I've did a `rg` on the code base and found more usages without `freq` in:

* test/model/test_forecast.py
* examples/anomaly_detection.py
* src/gluonts/nursery/autogluon_tabular/predictor.py

Can you also include them in this PR?

**Comment by kashif** [comment]
sure let me check

File: examples/anomaly_detection.py [hunk 2]
```diff
@@ -100,6 +100,7 @@
         time_index = pd.period_range(
             data_entry["forecast_start"][i],
             periods=context_length + prediction_length,
+            freq=data_entry["forecast_start"][i].freq,
```
Comment (jaheba) [inline examples/anomaly_detection.py:103]: Boy looks this ugly. But it is what it is.

**Review by jaheba — DISMISSED** [review]

File: examples/anomaly_detection.py (hunk 2 above)
Comment (kashif) [reply examples/anomaly_detection.py:103]: 👹

**Review by jaheba — APPROVED** [review]
